presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.reader;
+
+import com.facebook.presto.orc.OrcCorruptionException;
+import com.facebook.presto.orc.StreamDescriptor;
+import com.facebook.presto.orc.stream.InputStreamSource;
+import com.facebook.presto.orc.stream.InputStreamSources;
+import com.facebook.presto.orc.stream.LongInputStream;
+
+import java.io.IOException;
+
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.DICTIONARY_DATA;
+import static java.util.Objects.requireNonNull;
+
+public class LongDictionaryProvider
+{
+    private final InputStreamSources dictionaryStreamSources;
+
+    public static class GetDictionaryResult
+    {
+        private final long[] dictionaryBuffer;
+        private final boolean isNewBuffer;
+
+        GetDictionaryResult(long[] dictionaryBuffer, boolean isNewBuffer)
+        {
+            this.dictionaryBuffer = requireNonNull(dictionaryBuffer);
+            this.isNewBuffer = isNewBuffer;
+        }
+
+        public long[] getDictionaryBuffer()
+        {
+            return dictionaryBuffer;
+        }
+
+        public boolean getIsNewBuffer()
+        {
+            return isNewBuffer;
+        }
```

Comment:
If this is only used for memory accounting, maybe a better way is to use
```java
public long getRetainedSizeInBytes()
{
    return isNewBuffer ? sizeOf(dictionaryBuffer) : 0;
}
```
so we don't have to write `if` outside of the function.